tensor2tensor/layers/bayes.py

```diff
@@ -140,12 +143,11 @@ def __init__(self, mean=0., stddev=1.):
     self.stddev = stddev
 
   def __call__(self, x):
-    mean, stddev = x._parameters  # pylint: disable=protected-access
-    variance2 = tf.square(self.stddev)
-    variance_ratio = tf.square(stddev) / variance2
-    regularization = tf.square(mean - self.mean) / (2. * variance2)
-    regularization += (variance_ratio - 1. - tf.log(variance_ratio)) / 2.
-    return regularization
+    """Computes regularization given an ed.Normal random variable as input."""
+    if not isinstance(x, ed.RandomVariable):
+      raise ValueError('Input must be an ed.RandomVariable.')
+    random_variable = ed.Normal(loc=self.mean, scale=self.stddev)
```

Comment:
Any reason to not create this object in `__ini__` already instead of saving `self.mean` and `save.stddev` separately?

Reply:
Breaks Eager unfortunately if `self.mean` and `self.stddev` depend on trainable variables. See the above comment about caching. It would be nice if Distribution `__init__`s don't add TF ops to the graph.